src/content/docs/bff/fundamentals/blazor/index.md

```diff
@@ -2,91 +2,389 @@
+:::
+
+The Interactive Server mode allows developers to render pages built with Blazor components, and that also allows for interactivity. This mode is ideal for applications that require a rich user experience, such as a web application that allows users to create, edit, and delete data. The interactivity for this mode is handled by the Blazor Server framework powered by WebSockets and more specifically [SignalR](https://dotnet.microsoft.com/en-us/apps/aspnet/signalr).
+
+The BFF pattern is not typically applicable to this mode, as most interactivity is handled on the server by the Blazor Server framework with state changes being pushed to the client via WebSockets.
+
+You may still want to explore `AuthenticationStateProvider` for managing authentication state, see the section below for more information.
+
+### Interactive WebAssembly
 
-## Adding BFF To A Blazor Application
+:::note
+**We recommend using the BFF pattern with this rendering mode, as your frontend will be operating with the context of the client, and not the server.**
+:::
 
-Adding 
+The Interactive WebAssembly mode allows developers to render pages built with Blazor components, and that also allows for interactivity. This mode is ideal for applications that require a rich user experience, such as a web application that allows users to create, edit, and delete data. **The interactivity for this mode is handled by the Blazor WebAssembly framework and operates within the context of the client.**
+
+In a typical Blazor WebAssembly application, you will have three projects: `Client`, `Server`, and `Shared`. The `Client` project is the Blazor application that is rendered by the browser. The `Server` project is the ASP.NET Core web application that hosts the Blazor application. The `Shared` project is a project that contains C# classes that are shared between the `Client` and `Server` projects.
+
+Let's take a look at how to install and configure the BFF pattern given the above project structure.
+
+In the `Server` project, you will need to add the following NuGet packages, assuming you will want to use the OpenID Connect handler:
+
+```bash
+dotnet add package Duende.Bff
+dotnet add package Duende.Bff.Blazor
+dotnet add package Microsoft.AspNetCore.Authentication.OpenIdConnect
+```
+
+You will also need to modify the `Program.cs` file in the `Server` project to configure the BFF in the services collection:
 
 ```csharp
+// Server/Program.cs
 builder.Services.AddBff()
     .AddServerSideSessions() // Add in-memory implementation of server side sessions
     .AddBlazorServer();
+```
 
+The `AddBlazorServer` method will configure the BFF to use services on the host that allow the client to interact with the server securely.
 
-// ... <snip> ...
+You will also need to modify the ASP.NET Core pipeline to use the BFF:
```

Comment:
Tip: you can highlight the relevant lines in the snippet if you want https://starlight.astro.build/guides/authoring-content/#text-markers